src/serde_helpers.rs

```diff
@@ -70,6 +70,40 @@ pub mod object_id {
             Ok(oid.to_hex())
         }
     );
+
+    pub(crate) const OID_BYTES_NEWTYPE: &str = "$__bson_private_oid_bytes";
+    /// When serializing to a non-human-readable destination, will represent the [`ObjectId`] as a
```

Comment:
If we eliminated the human-readable-specific logic here, could users get the same effect with:

```rust
#[serde_as(as = "IfIsHumanReadable<serde_with::Same, object_id::AsBytes>")]
oid: ObjectId,
```
?
I'd prefer not to be opinionated about human-readable in these helpers if we can avoid it.